src/back/glsl/mod.rs

```diff
@@ -2534,6 +2534,13 @@ impl<'a, W: Write> Writer<'a, W> {
                     crate::ImageDimension::D3 => 3,
                     crate::ImageDimension::Cube => 2,
                 };
+
+                if components == 1 {
+                    write!(self.out, "uint(")?;
+                } else {
+                    write!(self.out, "uvec{components}(")?;
```

Comment:
The `spv` backend includes `array_coords` in determining the vector's size.
Not sure if this is need here as well.

Reply:
If the image is arrayed this would require adding one more component to the resulting vector, but this is only for queries of the size, all other queries return a scalar and shouldn't be splatted.

Reply:
Okay, should be corrected, but feel free to make any changes if needed.